wordpress-mobile/WordPress-Android
# Jetpack Kotlinization:  JetpackConnectionSource

Parent: #14904 

This PR converts `JetpackConnectionSource` from java to Kotlin
I also added a test, but it might very well be overkill. :)

To test:
- Launch the App
- Login with a site that doesn't have Jetpack installed (I used a pressable site)
- Navigate to stats
- The Jetpack required view should be shown
- Follow the prompts to install Jetpack 
- Follow the prompts for connecting to WP
- Enter valid credentials
- When prompted, allow connection
- App should load as normal

## Regression Notes
1. Potential unintended areas of impact
Jetpack Connections

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test a Jetpack connection

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
